gnocchi/rest/api.py

```diff
@@ -1135,6 +1135,8 @@ def post(self):
             abort(400, six.text_type(e))
         except indexer.ResourceAlreadyExists as e:
             abort(409, six.text_type(e))
+        except indexer.UnexpectedResourceTypeState as e:
```

Comment:
In case of e.state is "creating" we can retry create resource or maybe raise 409 also

Reply:
Do we want to do that server side? I'm not sure. Or maybe I'm lazy.

Reply:
i'm hesitant about 503. when i see 503, it's generally that service is completely offline/unaccessible.

Reply:
Right. Maybe 500 would be better.

Reply:
i think i agree with sileht, 5xx anything is very aggressive to me.

Reply:
I don't see the point of catching a 500 for an other 500.

Reply:
This one has a proper exception info returned?